### PR TITLE
DAOS-623 build: Use BUILD_TYPE=dev for travis build

### PR DESCRIPTION
**Pull request by jolivier23**
The java .xom file wants it in dev.   We should automate this but
this should fix the immediate issue.

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

**Review by frostedcmos — APPROVED** [review]

File: .travis.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ before_install:
 
 script:
  - docker build -f utils/docker/Dockerfile.$DOCKER_IMAGE -t daos/$DOCKER_IMAGE --build-arg NOBUILD=1 --build-arg UID=$UID .
- - docker run -v $PWD:/home/daos/daos:z daos/$DOCKER_IMAGE /bin/bash -c "scons --build-deps=yes install"
+ - docker run -v $PWD:/home/daos/daos:z daos/$DOCKER_IMAGE /bin/bash -c "scons --build-deps=yes install BUILD_TYPE=dev"
```
Comment (brianjmurrell) [inline .travis.yml:24]: Shouldn't this reflect dev or release, depending on whether the build is supposed to be a release build or not and be secure-by-default?

**Review by zjf2012 — APPROVED** [review]
approve this first. I'll provide the full solution in DAOS-6241.

**Comment by johannlombardi** [comment]
Let's land to fix the build failure and discuss a better solution address @brianjmurrell's feedback in a different PR.

**Comment by zjf2012** [comment]
> Let's land to fix the build failure and discuss a better solution address @brianjmurrell's feedback in a different PR.

yes. please check https://github.com/daos-stack/daos/pull/4038.

**Pull request merged**